data-prepper-plugins/opensearch-source/src/main/java/org/opensearch/dataprepper/plugins/source/opensearch/worker/client/ClusterClientFactory.java

```diff
@@ -5,6 +5,9 @@
 
 package org.opensearch.dataprepper.plugins.source.opensearch.worker.client;
 
-public interface ClusterClientFactory {
-    Object getClient();
+import org.opensearch.dataprepper.model.plugin.PluginComponentRefresher;
+import org.opensearch.dataprepper.plugins.source.opensearch.OpenSearchSourceConfiguration;
+
+public interface ClusterClientFactory<Client> {
+    PluginComponentRefresher<Client, OpenSearchSourceConfiguration> getClientRefresher();
```

Comment:
In line with my comment elsewhere, change this to:

```
Supplier<Client> getClientSupplier();
```

Reply:
Unfortunately there are some tight coupling in using APIs, e.g. OpenSearchIndexPartitionCreationSupplier needs the SearchAccessor to return more than Supplier. I improved it to be returning PluginComponentRefresher<Client, OpenSearchConfiguration>